[Feature] Pipe FileList

murphyatwork opened the pull request:
Fixes #issue

1. persist the file_list in OlapTable
2. provide `information_schema.pipes` and `information_schema.pipe_files` to view the status

## What type of PR is this:
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

murphyatwork committed:
support multiple files in path

Signed-off-by: Murphy <mofei@starrocks.com>

wanpengfei-git commented:
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 4 / 4 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/TableFunctionTable.java|4|4|100.00%|[]|

murphyatwork committed:
add CREATED_TIME in show pipes command

Signed-off-by: Murphy <mofei@starrocks.com>

murphyatwork committed:
support order by in show pipes command

Signed-off-by: Murphy <mofei@starrocks.com>

murphyatwork committed:
thrift interface

murphyatwork committed:
mock fe interface

Signed-off-by: Murphy <mofei@starrocks.com>

murphyatwork committed:
fix interface

murphyatwork committed:
implement pipes view

Signed-off-by: Murphy <mofei@starrocks.com>

murphyatwork committed:
implement pipe_files

murphyatwork committed:
add pipes view at information_schema

Signed-off-by: Murphy <mofei@starrocks.com>

murphyatwork committed:
implement pipes view in BE

Signed-off-by: Murphy <mofei@starrocks.com>

murphyatwork committed:
table repo

Signed-off-by: Murphy <mofei@starrocks.com>

wanpengfei-git commented:
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

murphyatwork committed:
do

Signed-off-by: Murphy <mofei@starrocks.com>

murphyatwork committed:
extract FileListRepo

Signed-off-by: Murphy <mofei@starrocks.com>

murphyatwork committed:
[Enhancement] improve error message of table function (#26861)

- Bucket error: `(1064, 'Access storage error. Error code: NoSuchBucket.
Error message: The specified bucket does not exist. ')`
- File path error: `(1064, 'Access storage error. Error message: no file
found with given path pattern: s3a://sdfsdf/')`
- Access Key error: `(1064, 'Access storage error. Error code:
InvalidAccessKeyId. Error message: The AWS Access Key Id you provided
does not exist in our records.. ')`

murphyatwork committed:
compile

Signed-off-by: Murphy <mofei@starrocks.com>

murphyatwork committed:
mock the pipe_files view

Signed-off-by: Murphy <mofei@starrocks.com>

murphyatwork committed:
parse json protocol

Signed-off-by: Murphy <mofei@starrocks.com>

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 62 / 133 (46.62%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|0|57|00.00%|[484, 485, 486, 487, 488, 489, 490, 491, 492, 493, 495, 496, 497, 498, 499, 500, 501, 502, 503, 504, 505, 507, 508, 509, 510, 511, 512, 513, 518, 519, 520, 521, 523, 524, 525, 526, 527, 529, 530, 533, 534, 535, 536, 537, 538, 539, 540, 542, 543, 545, 546, 550, 551, 552, 554, 556, 558]|
|:large_blue_circle:|com/starrocks/catalog/TableFunctionTable.java|0|3|00.00%|[155, 157, 158]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|4|7|57.14%|[3802, 3803, 3805]|
|:large_blue_circle:|com/starrocks/sql/ast/pipe/ShowPipeStmt.java|4|6|66.67%|[80, 84]|
|:large_blue_circle:|com/starrocks/catalog/CatalogUtils.java|2|3|66.67%|[52]|
|:large_blue_circle:|com/starrocks/load/pipe/PipeFile.java|5|7|71.43%|[60, 65]|
|:large_blue_circle:|com/starrocks/load/pipe/FileListRepo.java|12|15|80.00%|[57, 58, 59]|
|:large_blue_circle:|com/starrocks/catalog/Database.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/system/information/InfoSchemaDb.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/system/information/PipeFileSystemTable.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/PartitionInfo.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/FilePipeSource.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/Pipe.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PipeAnalyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|15|15|100.00%|[]|

murphyatwork committed:
change pipe_files meta

Signed-off-by: Murphy <mofei@starrocks.com>

murphyatwork committed:
modify pipe_files view

Signed-off-by: Murphy <mofei@starrocks.com>

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 69 / 140 (49.29%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|0|57|00.00%|[484, 485, 486, 487, 488, 489, 490, 491, 492, 493, 495, 496, 497, 498, 499, 500, 501, 502, 503, 504, 505, 507, 508, 509, 510, 511, 512, 513, 518, 519, 520, 521, 523, 524, 525, 526, 527, 529, 530, 533, 534, 535, 536, 537, 538, 539, 540, 542, 543, 545, 546, 550, 551, 552, 554, 556, 558]|
|:large_blue_circle:|com/starrocks/catalog/TableFunctionTable.java|0|3|00.00%|[155, 157, 158]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|4|7|57.14%|[3802, 3803, 3805]|
|:large_blue_circle:|com/starrocks/sql/ast/pipe/ShowPipeStmt.java|4|6|66.67%|[80, 84]|
|:large_blue_circle:|com/starrocks/catalog/CatalogUtils.java|2|3|66.67%|[52]|
|:large_blue_circle:|com/starrocks/load/pipe/PipeFile.java|5|7|71.43%|[60, 65]|
|:large_blue_circle:|com/starrocks/load/pipe/FileListRepo.java|12|15|80.00%|[57, 58, 59]|
|:large_blue_circle:|com/starrocks/catalog/Database.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/system/information/InfoSchemaDb.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/system/information/PipeFileSystemTable.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/PartitionInfo.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/FilePipeSource.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/Pipe.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PipeAnalyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|15|15|100.00%|[]|

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 79 / 149 (53.02%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/TableFunctionTable.java|0|3|00.00%|[155, 157, 158]|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|10|66|15.15%|[485, 486, 487, 488, 489, 490, 491, 492, 493, 495, 496, 497, 498, 499, 500, 501, 502, 503, 504, 505, 507, 508, 509, 510, 511, 512, 513, 518, 519, 520, 521, 523, 524, 525, 526, 527, 529, 530, 533, 534, 535, 536, 537, 538, 539, 540, 542, 543, 545, 546, 550, 551, 552, 554, 556, 558]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|4|7|57.14%|[3802, 3803, 3805]|
|:large_blue_circle:|com/starrocks/sql/ast/pipe/ShowPipeStmt.java|4|6|66.67%|[80, 84]|
|:large_blue_circle:|com/starrocks/catalog/CatalogUtils.java|2|3|66.67%|[52]|
|:large_blue_circle:|com/starrocks/load/pipe/PipeFile.java|5|7|71.43%|[60, 65]|
|:large_blue_circle:|com/starrocks/load/pipe/FileListRepo.java|12|15|80.00%|[57, 58, 59]|
|:large_blue_circle:|com/starrocks/catalog/Database.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/system/information/InfoSchemaDb.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/system/information/PipeFileSystemTable.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/PartitionInfo.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/FilePipeSource.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/Pipe.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PipeAnalyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|15|15|100.00%|[]|

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 79 / 149 (53.02%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/TableFunctionTable.java|0|3|00.00%|[155, 157, 158]|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|10|66|15.15%|[485, 486, 487, 488, 489, 490, 491, 492, 493, 495, 496, 497, 498, 499, 500, 501, 502, 503, 504, 505, 507, 508, 509, 510, 511, 512, 513, 518, 519, 520, 521, 523, 524, 525, 526, 527, 529, 530, 533, 534, 535, 536, 537, 538, 539, 540, 542, 543, 545, 546, 550, 551, 552, 554, 556, 558]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|4|7|57.14%|[3802, 3803, 3805]|
|:large_blue_circle:|com/starrocks/sql/ast/pipe/ShowPipeStmt.java|4|6|66.67%|[80, 84]|
|:large_blue_circle:|com/starrocks/catalog/CatalogUtils.java|2|3|66.67%|[52]|
|:large_blue_circle:|com/starrocks/load/pipe/PipeFile.java|5|7|71.43%|[60, 65]|
|:large_blue_circle:|com/starrocks/load/pipe/FileListRepo.java|12|15|80.00%|[57, 58, 59]|
|:large_blue_circle:|com/starrocks/catalog/Database.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/system/information/InfoSchemaDb.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/system/information/PipeFileSystemTable.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/PartitionInfo.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/FilePipeSource.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/Pipe.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PipeAnalyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|15|15|100.00%|[]|

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 69 / 138 (50.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|0|55|00.00%|[485, 486, 487, 488, 489, 490, 491, 492, 493, 495, 496, 497, 498, 499, 500, 501, 502, 503, 504, 505, 507, 508, 509, 510, 511, 512, 513, 518, 519, 520, 521, 523, 524, 525, 526, 527, 529, 530, 533, 534, 535, 536, 537, 538, 539, 540, 542, 543, 545, 546, 550, 551, 552, 554, 556]|
|:large_blue_circle:|com/starrocks/catalog/TableFunctionTable.java|0|3|00.00%|[155, 157, 158]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|4|7|57.14%|[3802, 3803, 3805]|
|:large_blue_circle:|com/starrocks/sql/ast/pipe/ShowPipeStmt.java|4|6|66.67%|[80, 84]|
|:large_blue_circle:|com/starrocks/catalog/CatalogUtils.java|2|3|66.67%|[52]|
|:large_blue_circle:|com/starrocks/load/pipe/PipeFile.java|5|7|71.43%|[60, 65]|
|:large_blue_circle:|com/starrocks/load/pipe/FileListRepo.java|12|15|80.00%|[57, 58, 59]|
|:large_blue_circle:|com/starrocks/catalog/Database.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/system/information/InfoSchemaDb.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/system/information/PipeFileSystemTable.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/PartitionInfo.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/FilePipeSource.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/Pipe.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PipeAnalyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|15|15|100.00%|[]|

murphyatwork committed:
parse record

Signed-off-by: Murphy <mofei@starrocks.com>

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 80 / 149 (53.69%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/TableFunctionTable.java|0|3|00.00%|[155, 157, 158]|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|3|60|05.00%|[484, 485, 486, 487, 488, 489, 490, 491, 492, 493, 495, 496, 497, 498, 499, 500, 501, 502, 503, 504, 505, 507, 508, 509, 510, 511, 512, 513, 518, 519, 520, 521, 523, 524, 525, 526, 527, 529, 530, 533, 534, 535, 536, 537, 538, 539, 540, 542, 543, 545, 546, 550, 551, 552, 554, 556, 558]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|4|7|57.14%|[3802, 3803, 3805]|
|:large_blue_circle:|com/starrocks/sql/ast/pipe/ShowPipeStmt.java|4|6|66.67%|[80, 84]|
|:large_blue_circle:|com/starrocks/catalog/CatalogUtils.java|2|3|66.67%|[52]|
|:large_blue_circle:|com/starrocks/load/pipe/FileListRepo.java|12|15|80.00%|[57, 58, 59]|
|:large_blue_circle:|com/starrocks/catalog/Database.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/PipeManager.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/system/information/InfoSchemaDb.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/PipeFile.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/util/DateUtils.java|8|8|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/system/information/PipeFileSystemTable.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/PartitionInfo.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/FilePipeSource.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/Pipe.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PipeAnalyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|15|15|100.00%|[]|

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 80 / 149 (53.69%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/TableFunctionTable.java|0|3|00.00%|[155, 157, 158]|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|3|60|05.00%|[484, 485, 486, 487, 488, 489, 490, 491, 492, 493, 495, 496, 497, 498, 499, 500, 501, 502, 503, 504, 505, 507, 508, 509, 510, 511, 512, 513, 518, 519, 520, 521, 523, 524, 525, 526, 527, 529, 530, 533, 534, 535, 536, 537, 538, 539, 540, 542, 543, 545, 546, 550, 551, 552, 554, 556, 558]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|4|7|57.14%|[3802, 3803, 3805]|
|:large_blue_circle:|com/starrocks/sql/ast/pipe/ShowPipeStmt.java|4|6|66.67%|[80, 84]|
|:large_blue_circle:|com/starrocks/catalog/CatalogUtils.java|2|3|66.67%|[52]|
|:large_blue_circle:|com/starrocks/load/pipe/FileListRepo.java|12|15|80.00%|[57, 58, 59]|
|:large_blue_circle:|com/starrocks/catalog/Database.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/PipeManager.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/system/information/InfoSchemaDb.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/PipeFile.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/util/DateUtils.java|8|8|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/system/information/PipeFileSystemTable.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/PartitionInfo.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/FilePipeSource.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/Pipe.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PipeAnalyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|15|15|100.00%|[]|

murphyatwork committed:
implement CRUD for FileListRepo

Signed-off-by: Murphy <mofei@starrocks.com>

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 82 / 151 (54.30%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/TableFunctionTable.java|0|3|00.00%|[155, 157, 158]|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|3|60|05.00%|[484, 485, 486, 487, 488, 489, 490, 491, 492, 493, 495, 496, 497, 498, 499, 500, 501, 502, 503, 504, 505, 507, 508, 509, 510, 511, 512, 513, 518, 519, 520, 521, 523, 524, 525, 526, 527, 529, 530, 533, 534, 535, 536, 537, 538, 539, 540, 542, 543, 545, 546, 550, 551, 552, 554, 556, 558]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|4|7|57.14%|[3802, 3803, 3805]|
|:large_blue_circle:|com/starrocks/sql/ast/pipe/ShowPipeStmt.java|4|6|66.67%|[80, 84]|
|:large_blue_circle:|com/starrocks/catalog/CatalogUtils.java|2|3|66.67%|[52]|
|:large_blue_circle:|com/starrocks/load/pipe/FileListRepo.java|12|15|80.00%|[57, 58, 59]|
|:large_blue_circle:|com/starrocks/catalog/Database.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/PipeManager.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/system/information/InfoSchemaDb.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/PipeFile.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/util/DateUtils.java|8|8|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/system/information/PipeFileSystemTable.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/PartitionInfo.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/FilePipeSource.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/Pipe.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PipeAnalyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|15|15|100.00%|[]|

murphyatwork committed:
fix

Signed-off-by: Murphy <mofei@starrocks.com>

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 82 / 151 (54.30%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/TableFunctionTable.java|0|3|00.00%|[155, 157, 158]|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|3|60|05.00%|[484, 485, 486, 487, 488, 489, 490, 491, 492, 493, 495, 496, 497, 498, 499, 500, 501, 502, 503, 504, 505, 507, 508, 509, 510, 511, 512, 513, 518, 519, 520, 521, 523, 524, 525, 526, 527, 529, 530, 533, 534, 535, 536, 537, 538, 539, 540, 542, 543, 545, 546, 550, 551, 552, 554, 556, 558]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|4|7|57.14%|[3802, 3803, 3805]|
|:large_blue_circle:|com/starrocks/sql/ast/pipe/ShowPipeStmt.java|4|6|66.67%|[80, 84]|
|:large_blue_circle:|com/starrocks/catalog/CatalogUtils.java|2|3|66.67%|[52]|
|:large_blue_circle:|com/starrocks/load/pipe/FileListRepo.java|12|15|80.00%|[57, 58, 59]|
|:large_blue_circle:|com/starrocks/catalog/Database.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/PipeManager.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/system/information/InfoSchemaDb.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/PipeFile.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/util/DateUtils.java|8|8|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/system/information/PipeFileSystemTable.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/PartitionInfo.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/FilePipeSource.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/Pipe.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PipeAnalyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|15|15|100.00%|[]|

murphyatwork committed:
fix sql bug

Signed-off-by: Murphy <mofei@starrocks.com>

murphyatwork committed:
filter staged files

Signed-off-by: Murphy <mofei@starrocks.com>

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 93 / 167 (55.69%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/TableFunctionTable.java|0|3|00.00%|[155, 157, 158]|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|3|60|05.00%|[484, 485, 486, 487, 488, 489, 490, 491, 492, 493, 495, 496, 497, 498, 499, 500, 501, 502, 503, 504, 505, 507, 508, 509, 510, 511, 512, 513, 518, 519, 520, 521, 523, 524, 525, 526, 527, 529, 530, 533, 534, 535, 536, 537, 538, 539, 540, 542, 543, 545, 546, 550, 551, 552, 554, 556, 558]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|4|7|57.14%|[3802, 3803, 3805]|
|:large_blue_circle:|com/starrocks/sql/ast/pipe/ShowPipeStmt.java|4|6|66.67%|[80, 84]|
|:large_blue_circle:|com/starrocks/catalog/CatalogUtils.java|2|3|66.67%|[52]|
|:large_blue_circle:|com/starrocks/load/pipe/FilePipeSource.java|12|16|75.00%|[71, 72, 125, 126]|
|:large_blue_circle:|com/starrocks/load/pipe/FileListRepo.java|13|17|76.47%|[57, 58, 59, 60]|
|:large_blue_circle:|com/starrocks/catalog/Database.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/PipeManager.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/system/information/InfoSchemaDb.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/PipeFile.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/util/DateUtils.java|8|8|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/system/information/PipeFileSystemTable.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/PartitionInfo.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/Pipe.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PipeAnalyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|15|15|100.00%|[]|

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 93 / 167 (55.69%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/TableFunctionTable.java|0|3|00.00%|[155, 157, 158]|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|3|60|05.00%|[484, 485, 486, 487, 488, 489, 490, 491, 492, 493, 495, 496, 497, 498, 499, 500, 501, 502, 503, 504, 505, 507, 508, 509, 510, 511, 512, 513, 518, 519, 520, 521, 523, 524, 525, 526, 527, 529, 530, 533, 534, 535, 536, 537, 538, 539, 540, 542, 543, 545, 546, 550, 551, 552, 554, 556, 558]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|4|7|57.14%|[3802, 3803, 3805]|
|:large_blue_circle:|com/starrocks/sql/ast/pipe/ShowPipeStmt.java|4|6|66.67%|[80, 84]|
|:large_blue_circle:|com/starrocks/catalog/CatalogUtils.java|2|3|66.67%|[52]|
|:large_blue_circle:|com/starrocks/load/pipe/FilePipeSource.java|12|16|75.00%|[71, 72, 125, 126]|
|:large_blue_circle:|com/starrocks/load/pipe/FileListRepo.java|13|17|76.47%|[57, 58, 59, 60]|
|:large_blue_circle:|com/starrocks/catalog/Database.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/PipeManager.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/system/information/InfoSchemaDb.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/PipeFile.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/util/DateUtils.java|8|8|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/system/information/PipeFileSystemTable.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/PartitionInfo.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/Pipe.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PipeAnalyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|15|15|100.00%|[]|

murphyatwork committed:
refactor FileRecord

Signed-off-by: Murphy <mofei@starrocks.com>

murphyatwork committed:
fix state updating

Signed-off-by: Murphy <mofei@starrocks.com>

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 102 / 193 (52.85%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/TableFunctionTable.java|0|3|00.00%|[155, 157, 158]|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsFsManager.java|0|13|00.00%|[1186, 1187, 1188, 1189, 1191, 1192, 1193, 1194, 1196, 1197, 1198, 1199, 1200]|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsService.java|0|5|00.00%|[72, 73, 74, 78, 80]|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|3|59|05.08%|[485, 486, 487, 488, 489, 490, 491, 492, 493, 495, 496, 497, 498, 499, 500, 501, 502, 503, 504, 505, 507, 508, 509, 510, 511, 512, 513, 518, 519, 520, 521, 523, 524, 525, 526, 527, 529, 530, 533, 534, 535, 536, 537, 538, 539, 540, 542, 543, 545, 546, 550, 551, 552, 554, 556, 558]|
|:large_blue_circle:|com/starrocks/load/pipe/FilePipePiece.java|1|2|50.00%|[31]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|4|7|57.14%|[3802, 3803, 3805]|
|:large_blue_circle:|com/starrocks/sql/ast/pipe/ShowPipeStmt.java|4|6|66.67%|[80, 84]|
|:large_blue_circle:|com/starrocks/catalog/CatalogUtils.java|2|3|66.67%|[52]|
|:large_blue_circle:|com/starrocks/load/pipe/FileListRepo.java|9|12|75.00%|[57, 58, 60]|
|:large_blue_circle:|com/starrocks/fs/HdfsUtil.java|6|7|85.71%|[94]|
|:large_blue_circle:|com/starrocks/load/pipe/FilePipeSource.java|18|21|85.71%|[71, 72, 129]|
|:large_blue_circle:|com/starrocks/catalog/Database.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/PipeManager.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/system/information/InfoSchemaDb.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/util/DateUtils.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/system/information/PipeFileSystemTable.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/PartitionInfo.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/Pipe.java|11|11|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PipeAnalyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|15|15|100.00%|[]|

murphyatwork committed:
test case

Signed-off-by: Murphy <mofei@starrocks.com>

murphyatwork committed:
rename alter pipe pause to suspend

Signed-off-by: Murphy <mofei@starrocks.com>

murphyatwork committed:
fix codestyle

Signed-off-by: Murphy <mofei@starrocks.com>

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 109 / 202 (53.96%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/TableFunctionTable.java|0|3|00.00%|[153, 154, 155]|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsFsManager.java|0|13|00.00%|[1186, 1187, 1188, 1189, 1191, 1192, 1193, 1194, 1196, 1197, 1198, 1199, 1200]|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsService.java|0|5|00.00%|[72, 73, 74, 78, 80]|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|3|59|05.08%|[485, 486, 487, 488, 489, 490, 491, 492, 493, 495, 496, 497, 498, 499, 500, 501, 502, 503, 504, 505, 507, 508, 509, 510, 511, 512, 513, 518, 519, 520, 521, 523, 524, 525, 526, 527, 529, 530, 533, 534, 535, 536, 537, 538, 539, 540, 542, 543, 545, 546, 550, 551, 552, 554, 556, 558]|
|:large_blue_circle:|com/starrocks/load/pipe/FilePipePiece.java|1|2|50.00%|[31]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|4|7|57.14%|[3802, 3803, 3805]|
|:large_blue_circle:|com/starrocks/sql/ast/pipe/ShowPipeStmt.java|4|6|66.67%|[80, 84]|
|:large_blue_circle:|com/starrocks/catalog/CatalogUtils.java|2|3|66.67%|[52]|
|:large_blue_circle:|com/starrocks/load/pipe/FileListRepo.java|9|12|75.00%|[57, 58, 60]|
|:large_blue_circle:|com/starrocks/fs/HdfsUtil.java|6|7|85.71%|[94]|
|:large_blue_circle:|com/starrocks/load/pipe/FilePipeSource.java|18|21|85.71%|[71, 72, 129]|
|:large_blue_circle:|com/starrocks/load/pipe/Pipe.java|12|14|85.71%|[374, 379]|
|:large_blue_circle:|com/starrocks/catalog/Database.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/PipeManager.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/system/information/InfoSchemaDb.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/pipe/AlterPipePauseResume.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/util/DateUtils.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/system/information/PipeFileSystemTable.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/PartitionInfo.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PipeAnalyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|15|15|100.00%|[]|

murphyatwork committed:
use etag as file version

Signed-off-by: Murphy <mofei@starrocks.com>

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 109 / 202 (53.96%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/TableFunctionTable.java|0|3|00.00%|[153, 154, 155]|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsFsManager.java|0|13|00.00%|[1186, 1187, 1188, 1189, 1191, 1192, 1193, 1194, 1196, 1197, 1198, 1199, 1200]|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsService.java|0|5|00.00%|[72, 73, 74, 78, 80]|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|3|59|05.08%|[485, 486, 487, 488, 489, 490, 491, 492, 493, 495, 496, 497, 498, 499, 500, 501, 502, 503, 504, 505, 507, 508, 509, 510, 511, 512, 513, 518, 519, 520, 521, 523, 524, 525, 526, 527, 529, 530, 533, 534, 535, 536, 537, 538, 539, 540, 542, 543, 545, 546, 550, 551, 552, 554, 556, 558]|
|:large_blue_circle:|com/starrocks/load/pipe/FilePipePiece.java|1|2|50.00%|[31]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|4|7|57.14%|[3802, 3803, 3805]|
|:large_blue_circle:|com/starrocks/sql/ast/pipe/ShowPipeStmt.java|4|6|66.67%|[80, 84]|
|:large_blue_circle:|com/starrocks/catalog/CatalogUtils.java|2|3|66.67%|[52]|
|:large_blue_circle:|com/starrocks/load/pipe/FileListRepo.java|9|12|75.00%|[57, 58, 60]|
|:large_blue_circle:|com/starrocks/fs/HdfsUtil.java|6|7|85.71%|[94]|
|:large_blue_circle:|com/starrocks/load/pipe/FilePipeSource.java|18|21|85.71%|[71, 72, 129]|
|:large_blue_circle:|com/starrocks/load/pipe/Pipe.java|12|14|85.71%|[374, 379]|
|:large_blue_circle:|com/starrocks/catalog/Database.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/PipeManager.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/system/information/InfoSchemaDb.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/pipe/AlterPipePauseResume.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/util/DateUtils.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/system/information/PipeFileSystemTable.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/PartitionInfo.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PipeAnalyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|15|15|100.00%|[]|

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 109 / 202 (53.96%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/TableFunctionTable.java|0|3|00.00%|[153, 154, 155]|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsFsManager.java|0|13|00.00%|[1186, 1187, 1188, 1189, 1191, 1192, 1193, 1194, 1196, 1197, 1198, 1199, 1200]|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsService.java|0|5|00.00%|[72, 73, 74, 78, 80]|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|3|59|05.08%|[485, 486, 487, 488, 489, 490, 491, 492, 493, 495, 496, 497, 498, 499, 500, 501, 502, 503, 504, 505, 507, 508, 509, 510, 511, 512, 513, 518, 519, 520, 521, 523, 524, 525, 526, 527, 529, 530, 533, 534, 535, 536, 537, 538, 539, 540, 542, 543, 545, 546, 550, 551, 552, 554, 556, 558]|
|:large_blue_circle:|com/starrocks/load/pipe/FilePipePiece.java|1|2|50.00%|[31]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|4|7|57.14%|[3802, 3803, 3805]|
|:large_blue_circle:|com/starrocks/sql/ast/pipe/ShowPipeStmt.java|4|6|66.67%|[80, 84]|
|:large_blue_circle:|com/starrocks/catalog/CatalogUtils.java|2|3|66.67%|[52]|
|:large_blue_circle:|com/starrocks/load/pipe/FileListRepo.java|9|12|75.00%|[57, 58, 60]|
|:large_blue_circle:|com/starrocks/fs/HdfsUtil.java|6|7|85.71%|[94]|
|:large_blue_circle:|com/starrocks/load/pipe/FilePipeSource.java|18|21|85.71%|[71, 72, 129]|
|:large_blue_circle:|com/starrocks/load/pipe/Pipe.java|12|14|85.71%|[374, 379]|
|:large_blue_circle:|com/starrocks/catalog/Database.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/PipeManager.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/system/information/InfoSchemaDb.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/pipe/AlterPipePauseResume.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/util/DateUtils.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/system/information/PipeFileSystemTable.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/PartitionInfo.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PipeAnalyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|15|15|100.00%|[]|

imay commented on fe/fe-core/src/main/java/com/starrocks/catalog/system/information/PipesSystemTable.java:
```diff
@@ -0,0 +1,38 @@
+// Licensed to the Apache Software Foundation (ASF) under one
```
wrong license header, please use starrocks license header

mofeiatwork replied:
fixed

rickif commented on be/src/exec/schema_scanner/schema_pipe_files.cpp:
```diff
@@ -26,47 +27,88 @@ SchemaScanner::ColumnDesc SchemaTablePipeFiles::_s_columns[] = {
+        : SchemaScanner(_s_columns, sizeof(_s_columns) / sizeof(SchemaScanner::ColumnDesc)) {}
 
 Status SchemaTablePipeFiles::start(RuntimeState* state) {
     RETURN_IF_ERROR(SchemaScanner::start(state));
```
```suggestion
    return SchemaScanner::start(state);
```

rickif commented on be/src/exec/schema_scanner/schema_pipes.cpp:
```diff
@@ -0,0 +1,88 @@
+        : SchemaScanner(_s_columns, sizeof(_s_columns) / sizeof(SchemaScanner::ColumnDesc)) {}
+
+Status SchemaTablePipes::start(RuntimeState* state) {
+    RETURN_IF_ERROR(SchemaScanner::start(state));
```
```suggestion
    return SchemaScanner::start(state);
```

murphyatwork committed:
fix format

Signed-off-by: Murphy <mofei@starrocks.com>

murphyatwork committed:
fix copyright

Signed-off-by: Murphy <mofei@starrocks.com>

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 115 / 204 (56.37%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/TableFunctionTable.java|0|3|00.00%|[153, 154, 155]|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsFsManager.java|0|13|00.00%|[1186, 1187, 1188, 1189, 1191, 1192, 1193, 1194, 1196, 1197, 1198, 1199, 1200]|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsService.java|0|5|00.00%|[72, 73, 74, 78, 80]|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|3|59|05.08%|[485, 486, 487, 488, 489, 490, 491, 492, 493, 495, 496, 497, 498, 499, 500, 501, 502, 503, 504, 505, 507, 508, 509, 510, 511, 512, 513, 518, 519, 520, 521, 523, 524, 525, 526, 527, 529, 530, 533, 534, 535, 536, 537, 538, 539, 540, 542, 543, 545, 546, 550, 551, 552, 554, 556, 558]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|4|7|57.14%|[3802, 3803, 3805]|
|:large_blue_circle:|com/starrocks/sql/ast/pipe/ShowPipeStmt.java|4|6|66.67%|[80, 84]|
|:large_blue_circle:|com/starrocks/catalog/CatalogUtils.java|2|3|66.67%|[52]|
|:large_blue_circle:|com/starrocks/load/pipe/FileListRepo.java|10|14|71.43%|[57, 59, 60, 69]|
|:large_blue_circle:|com/starrocks/fs/HdfsUtil.java|6|7|85.71%|[94]|
|:large_blue_circle:|com/starrocks/load/pipe/FilePipeSource.java|18|19|94.74%|[71]|
|:large_blue_circle:|com/starrocks/catalog/Database.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/PipeManager.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/system/information/InfoSchemaDb.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/pipe/AlterPipePauseResume.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/util/DateUtils.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/system/information/PipeFileSystemTable.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/PartitionInfo.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/Pipe.java|16|16|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PipeAnalyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|15|15|100.00%|[]|

murphyatwork committed:
fix comment

Signed-off-by: Murphy <mofei@starrocks.com>

wanpengfei-git commented:
#### [FE PR Coverage Check]

coverage check fail. please retry. :fearful:

[Please let me know](https://github.com/naver/cover-checker/issues/new) when error again.

java.io.IOException: Server returned HTTP response code: 500 for URL: https://api.github.com/repos/StarRocks/starrocks/issues/26197/comments

murphyatwork committed:
fix test

Signed-off-by: Murphy <mofei@starrocks.com>

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 112 / 199 (56.28%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/TableFunctionTable.java|0|3|00.00%|[153, 154, 155]|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsFsManager.java|0|13|00.00%|[1186, 1187, 1188, 1189, 1191, 1192, 1193, 1194, 1196, 1197, 1198, 1199, 1200]|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsService.java|0|5|00.00%|[72, 73, 74, 78, 80]|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|3|59|05.08%|[485, 486, 487, 488, 489, 490, 491, 492, 493, 495, 496, 497, 498, 499, 500, 501, 502, 503, 504, 505, 507, 508, 509, 510, 511, 512, 513, 518, 519, 520, 521, 523, 524, 525, 526, 527, 529, 530, 533, 534, 535, 536, 537, 538, 539, 540, 542, 543, 545, 546, 550, 551, 552, 554, 556, 558]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|4|7|57.14%|[3802, 3803, 3805]|
|:large_blue_circle:|com/starrocks/sql/ast/pipe/ShowPipeStmt.java|4|6|66.67%|[80, 84]|
|:large_blue_circle:|com/starrocks/catalog/CatalogUtils.java|2|3|66.67%|[52]|
|:large_blue_circle:|com/starrocks/fs/HdfsUtil.java|6|7|85.71%|[94]|
|:large_blue_circle:|com/starrocks/load/pipe/FilePipeSource.java|18|21|85.71%|[71, 72, 129]|
|:large_blue_circle:|com/starrocks/catalog/Database.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/PipeManager.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/system/information/InfoSchemaDb.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/pipe/AlterPipePauseResume.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/util/DateUtils.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/system/information/PipeFileSystemTable.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsFs.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/PartitionInfo.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/Pipe.java|16|16|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PipeAnalyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|15|15|100.00%|[]|

murphyatwork committed:
fix comment

Signed-off-by: Murphy <mofei@starrocks.com>

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 112 / 199 (56.28%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/TableFunctionTable.java|0|3|00.00%|[153, 154, 155]|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsFsManager.java|0|13|00.00%|[1186, 1187, 1188, 1189, 1191, 1192, 1193, 1194, 1196, 1197, 1198, 1199, 1200]|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsService.java|0|5|00.00%|[72, 73, 74, 78, 80]|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|3|59|05.08%|[485, 486, 487, 488, 489, 490, 491, 492, 493, 495, 496, 497, 498, 499, 500, 501, 502, 503, 504, 505, 507, 508, 509, 510, 511, 512, 513, 518, 519, 520, 521, 523, 524, 525, 526, 527, 529, 530, 533, 534, 535, 536, 537, 538, 539, 540, 542, 543, 545, 546, 550, 551, 552, 554, 556, 558]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|4|7|57.14%|[3802, 3803, 3805]|
|:large_blue_circle:|com/starrocks/sql/ast/pipe/ShowPipeStmt.java|4|6|66.67%|[80, 84]|
|:large_blue_circle:|com/starrocks/catalog/CatalogUtils.java|2|3|66.67%|[52]|
|:large_blue_circle:|com/starrocks/fs/HdfsUtil.java|6|7|85.71%|[94]|
|:large_blue_circle:|com/starrocks/load/pipe/FilePipeSource.java|18|21|85.71%|[71, 72, 129]|
|:large_blue_circle:|com/starrocks/catalog/Database.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/PipeManager.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/system/information/InfoSchemaDb.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/pipe/AlterPipePauseResume.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/util/DateUtils.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/system/information/PipeFileSystemTable.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsFs.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/PartitionInfo.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/Pipe.java|16|16|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PipeAnalyzer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|15|15|100.00%|[]|

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 112 / 180 (62.22%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/TableFunctionTable.java|0|3|00.00%|[153, 154, 155]|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsFsManager.java|0|13|00.00%|[1186, 1187, 1188, 1189, 1191, 1192, 1193, 1194, 1196, 1197, 1198, 1199, 1200]|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsService.java|0|5|00.00%|[72, 73, 74, 78, 80]|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|16|55|29.09%|[507, 508, 509, 510, 511, 512, 513, 518, 519, 520, 521, 523, 524, 525, 526, 527, 529, 530, 533, 534, 535, 536, 537, 538, 539, 540, 542, 543, 545, 546, 550, 551, 552, 554, 556, 558, 578, 579, 582]|
|:large_blue_circle:|com/starrocks/catalog/Database.java|2|3|66.67%|[615]|
|:large_blue_circle:|com/starrocks/sql/ast/pipe/ShowPipeStmt.java|4|6|66.67%|[80, 84]|
|:large_blue_circle:|com/starrocks/catalog/CatalogUtils.java|2|3|66.67%|[52]|
|:large_blue_circle:|com/starrocks/fs/HdfsUtil.java|6|7|85.71%|[94]|
|:large_blue_circle:|com/starrocks/load/pipe/FilePipeSource.java|18|21|85.71%|[71, 72, 129]|
|:large_blue_circle:|com/starrocks/load/pipe/PipeManager.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/system/information/InfoSchemaDb.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/pipe/AlterPipePauseResume.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/util/DateUtils.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/system/information/PipeFileSystemTable.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsFs.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/PartitionInfo.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/Pipe.java|16|16|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PipeAnalyzer.java|2|2|100.00%|[]|

banmoy commented on fe/fe-core/src/main/java/com/starrocks/load/pipe/filelist/FileListTableRepo.java:
```diff
@@ -0,0 +1,133 @@
+    protected static final String FILE_LIST_FULL_NAME = FILE_LIST_DB_NAME + "." + FILE_LIST_TABLE_NAME;
+
+    protected static final String FILE_LIST_TABLE_CREATE =
+            "CREATE TABLE IF NOT EXISTS %s (" +
```
why not includes file_rows and error fields?

mofeiatwork replied:
I have not implement these two functionalities, so I prefer to postpone adding these two columns

murphyatwork committed:
fix copyright

Signed-off-by: Murphy <mofei@starrocks.com>

mofeiatwork commented on fe/fe-core/src/main/java/com/starrocks/catalog/system/information/PipeFileSystemTable.java:
```diff
@@ -26,21 +26,26 @@ public class PipeFileSystemTable {
                         .column("DATABASE_NAME", ScalarType.createVarcharType(SystemTable.NAME_CHAR_LEN))
                         .column("PIPE_ID", ScalarType.BIGINT)
                         .column("PIPE_NAME", ScalarType.createVarcharType(SystemTable.NAME_CHAR_LEN))
+
```
yes. The first section is identifiers, the second section is properties of file, the third is state, the fourth is error message.

mofeiatwork commented on fe/fe-core/src/main/java/com/starrocks/load/pipe/filelist/FileListTableRepo.java:
```diff
@@ -0,0 +1,133 @@
+
+    private static final Logger LOG = LogManager.getLogger(FileListTableRepo.class);
+
+    protected static final String FILE_LIST_DB_NAME = StatsConstants.STATISTICS_DB_NAME;
```
- `file_list` is a hidden table, not visible to users.
- so I put this table in the `_statistics__` database, along with `column_statistics`, `table_statistic_v1`

wanpengfei-git commented:
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 429 / 527 (81.40%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsFs.java|0|2|00.00%|[40, 42]|
|:large_blue_circle:|com/starrocks/catalog/TableFunctionTable.java|0|4|00.00%|[153, 154, 155, 156]|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsService.java|0|5|00.00%|[73, 74, 75, 76, 78]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PipeAnalyzer.java|1|10|10.00%|[175, 176, 177, 178, 181, 182, 183, 184, 185]|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsFsManager.java|5|11|45.45%|[1192, 1193, 1194, 1195, 1196, 1197]|
|:large_blue_circle:|com/starrocks/load/pipe/FilePipePiece.java|1|2|50.00%|[31]|
|:large_blue_circle:|com/starrocks/common/util/DateUtils.java|3|5|60.00%|[93, 103]|
|:large_blue_circle:|com/starrocks/sql/ast/pipe/ShowPipeStmt.java|12|18|66.67%|[81, 88, 115, 122, 130, 131]|
|:large_blue_circle:|com/starrocks/load/pipe/filelist/FileListTableRepo.java|18|23|78.26%|[95, 110, 120, 129, 130]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|22|28|78.57%|[2664, 2669, 2674, 2675, 2676, 2689]|
|:large_blue_circle:|com/starrocks/fs/HdfsUtil.java|4|5|80.00%|[92]|
|:large_blue_circle:|com/starrocks/load/pipe/PipeFileRecord.java|77|95|81.05%|[81, 82, 83, 95, 96, 97, 98, 99, 100, 124, 125, 152, 160, 186, 206, 214, 218, 227]|
|:large_blue_circle:|com/starrocks/load/pipe/filelist/RepoAccessor.java|65|78|83.33%|[45, 46, 47, 69, 70, 71, 80, 81, 82, 105, 110, 111, 112]|
|:large_blue_circle:|com/starrocks/load/pipe/filelist/RepoExecutor.java|37|44|84.09%|[59, 60, 61, 62, 63, 70, 85]|
|:large_blue_circle:|com/starrocks/catalog/system/information/PipesSystemTable.java|7|8|87.50%|[23]|
|:large_blue_circle:|com/starrocks/load/pipe/FilePipeSource.java|21|24|87.50%|[71, 150, 151]|
|:large_blue_circle:|com/starrocks/load/pipe/filelist/RepoCreator.java|39|44|88.64%|[44, 45, 80, 81, 82]|
|:large_blue_circle:|com/starrocks/load/pipe/Pipe.java|16|18|88.89%|[143, 388]|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|55|57|96.49%|[560, 561]|
|:large_blue_circle:|com/starrocks/catalog/Database.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/PipeManager.java|19|19|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/filelist/FileListRepo.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/system/information/InfoSchemaDb.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/pipe/AlterPipePauseResume.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/system/information/PipeFileSystemTable.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/PartitionInfo.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/statistic/StatisticsMetaManager.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/CatalogUtils.java|1|1|100.00%|[]|

murphyatwork committed:
fix sqltest

wanpengfei-git commented:
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 440 / 527 (83.49%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsFs.java|0|2|00.00%|[40, 42]|
|:large_blue_circle:|com/starrocks/catalog/TableFunctionTable.java|0|4|00.00%|[153, 154, 155, 156]|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsService.java|0|5|00.00%|[73, 74, 75, 76, 78]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PipeAnalyzer.java|1|10|10.00%|[175, 176, 177, 178, 181, 182, 183, 184, 185]|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsFsManager.java|5|11|45.45%|[1192, 1193, 1194, 1195, 1196, 1197]|
|:large_blue_circle:|com/starrocks/load/pipe/FilePipePiece.java|1|2|50.00%|[31]|
|:large_blue_circle:|com/starrocks/common/util/DateUtils.java|3|5|60.00%|[93, 103]|
|:large_blue_circle:|com/starrocks/sql/ast/pipe/ShowPipeStmt.java|12|18|66.67%|[81, 88, 115, 122, 130, 131]|
|:large_blue_circle:|com/starrocks/load/pipe/filelist/FileListTableRepo.java|18|23|78.26%|[95, 110, 120, 129, 130]|
|:large_blue_circle:|com/starrocks/fs/HdfsUtil.java|4|5|80.00%|[92]|
|:large_blue_circle:|com/starrocks/load/pipe/PipeFileRecord.java|77|95|81.05%|[81, 82, 83, 95, 96, 97, 98, 99, 100, 124, 125, 152, 160, 186, 206, 214, 218, 227]|
|:large_blue_circle:|com/starrocks/load/pipe/filelist/RepoAccessor.java|65|78|83.33%|[45, 46, 47, 69, 70, 71, 80, 81, 82, 105, 110, 111, 112]|
|:large_blue_circle:|com/starrocks/catalog/system/information/PipesSystemTable.java|7|8|87.50%|[23]|
|:large_blue_circle:|com/starrocks/load/pipe/FilePipeSource.java|21|24|87.50%|[71, 150, 151]|
|:large_blue_circle:|com/starrocks/load/pipe/filelist/RepoCreator.java|39|44|88.64%|[44, 45, 80, 81, 82]|
|:large_blue_circle:|com/starrocks/load/pipe/Pipe.java|16|18|88.89%|[143, 388]|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|55|57|96.49%|[560, 561]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|27|28|96.43%|[2689]|
|:large_blue_circle:|com/starrocks/load/pipe/filelist/RepoExecutor.java|43|44|97.73%|[85]|
|:large_blue_circle:|com/starrocks/catalog/Database.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/PipeManager.java|19|19|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/filelist/FileListRepo.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/system/information/InfoSchemaDb.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/pipe/AlterPipePauseResume.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/system/information/PipeFileSystemTable.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/PartitionInfo.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/statistic/StatisticsMetaManager.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/CatalogUtils.java|1|1|100.00%|[]|

satanson commented on be/src/exec/schema_scanner/schema_helper.cpp:
```diff
@@ -56,6 +56,18 @@ Status SchemaHelper::list_materialized_view_status(const std::string& ip, const
+            ip, port, [&req, &res](FrontendServiceConnection& client) { client->listPipeFiles(*res, req); });
+}
+
 Status SchemaHelper::get_tables_info(const std::string& ip, const int32_t port, const TGetTablesInfoRequest& request,
```
L#73, show_varialbes typo

mofeiatwork replied:
fixed

satanson commented on be/src/exec/schema_scanner/schema_pipe_files.cpp:
```diff
@@ -26,47 +27,86 @@ SchemaScanner::ColumnDesc SchemaTablePipeFiles::_s_columns[] = {
+    auto& pipe_file = _pipe_files_result.pipe_files.at(_cur_row);
+    _cur_row++;
+    return {
+            Slice(pipe_file.database_name),
```
Slice excapes  outside, is it safe?  if _pipeline_files_result is resized or cleared?

mofeiatwork replied:
it's safe until now, since the RPC result is immutable after fired, and the `Scanner` only iterate the result row-wise.

satanson commented on be/src/exec/schema_scanner/schema_pipe_files.cpp:
```diff
@@ -26,47 +27,86 @@ SchemaScanner::ColumnDesc SchemaTablePipeFiles::_s_columns[] = {
 
-Status SchemaTablePipeFiles::fill_chunk(ChunkPtr* chunk) {
-    _cur_idx++;
+Status SchemaTablePipeFiles::_fill_chunk(ChunkPtr* chunk) {
```
the result chunk contains only one row?

mofeiatwork replied:
yes, `get_next` only return one row each time

satanson commented on be/src/exec/schema_scanner/schema_pipes.cpp:
```diff
@@ -0,0 +1,86 @@
+    return SchemaHelper::list_pipes(*(_param->ip), _param->port, params, &_pipes_result);
+}
+
+Status SchemaTablePipes::get_next(ChunkPtr* chunk, bool* eos) {
```
this function seems like SchemaTablePipeFiles::get_next,  can we extract the common part to form a function?

mofeiatwork replied:
good idea. I will do it in next PR

satanson commented on be/src/exec/schema_scanner/schema_pipes.cpp:
```diff
@@ -0,0 +1,86 @@
+    };
+}
+
+Status SchemaTablePipes::_fill_chunk(ChunkPtr* chunk) {
```
get_next only returns one row each time?

mofeiatwork replied:
yes, that's the semantic of `get_next`.

mofeiatwork commented on fe/fe-core/src/main/java/com/starrocks/load/pipe/filelist/FileListTableRepo.java:
```diff
@@ -0,0 +1,134 @@
+                    " start_load, finish_load";
+
+    protected static final String SELECT_FILES =
+            "SELECT " + ALL_COLUMNS + " FROM " + FILE_LIST_FULL_NAME;
```
it's impossible, because user does not have access to SQL, but I wrote all these sql.

but generally it's a good idea.

mofeiatwork replied:
fixed

mofeiatwork commented on fe/fe-core/src/main/java/com/starrocks/fs/hdfs/HdfsService.java:
```diff
@@ -67,6 +68,16 @@ public void listPath(TBrokerListPathRequest request, List<TBrokerFileStatus> fil
 
+    public List<FileStatus> listFileMeta(String path, Map<String, String> properties, boolean skipDir)
+            throws UserException {
+        LOG.info("try to list file meta: {}", path);
```
yes, every `HdfsService` interface would log the action.

murphyatwork committed:
quote sql identifier

Signed-off-by: Murphy <mofei@starrocks.com>

mofeiatwork commented on fe/fe-core/src/main/java/com/starrocks/load/pipe/filelist/RepoExecutor.java:
```diff
@@ -0,0 +1,112 @@
+            StatementBase parsedStmt = SqlParser.parseOneWithStarRocksDialect(sql, context.getSessionVariable());
+            Preconditions.checkState(parsedStmt instanceof DmlStmt, "the statement should be dml");
+            DmlStmt dmlStmt = (DmlStmt) parsedStmt;
+            ExecPlan execPlan = StatementPlanner.plan(parsedStmt, context, TResultSinkType.HTTP_PROTOCAL);
```
HTTP should be the general protocol if FE needs to process response of BE, so I choose it.

Currently I don't process the DML result of BE, but someday I should need it.

mofeiatwork commented on be/src/exec/schema_scanner/schema_pipes.cpp:
```diff
@@ -0,0 +1,86 @@
+}
+
+DatumArray SchemaTablePipes::_build_row() {
+    auto& pipe = _pipes_result.pipes.at(_cur_row);
```
fixed

murphyatwork committed:
fix comment

Signed-off-by: Murphy <mofei@starrocks.com>

wanpengfei-git commented:
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 440 / 527 (83.49%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsFs.java|0|2|00.00%|[40, 42]|
|:large_blue_circle:|com/starrocks/catalog/TableFunctionTable.java|0|4|00.00%|[153, 154, 155, 156]|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsService.java|0|5|00.00%|[73, 74, 75, 76, 78]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PipeAnalyzer.java|1|10|10.00%|[175, 176, 177, 178, 181, 182, 183, 184, 185]|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsFsManager.java|5|11|45.45%|[1192, 1193, 1194, 1195, 1196, 1197]|
|:large_blue_circle:|com/starrocks/load/pipe/FilePipePiece.java|1|2|50.00%|[31]|
|:large_blue_circle:|com/starrocks/common/util/DateUtils.java|3|5|60.00%|[93, 103]|
|:large_blue_circle:|com/starrocks/sql/ast/pipe/ShowPipeStmt.java|12|18|66.67%|[81, 88, 115, 122, 130, 131]|
|:large_blue_circle:|com/starrocks/load/pipe/filelist/FileListTableRepo.java|18|23|78.26%|[95, 110, 120, 129, 130]|
|:large_blue_circle:|com/starrocks/fs/HdfsUtil.java|4|5|80.00%|[92]|
|:large_blue_circle:|com/starrocks/load/pipe/PipeFileRecord.java|77|95|81.05%|[81, 82, 83, 95, 96, 97, 98, 99, 100, 124, 125, 152, 160, 186, 206, 214, 218, 227]|
|:large_blue_circle:|com/starrocks/load/pipe/filelist/RepoAccessor.java|65|78|83.33%|[45, 46, 47, 69, 70, 71, 80, 81, 82, 105, 110, 111, 112]|
|:large_blue_circle:|com/starrocks/catalog/system/information/PipesSystemTable.java|7|8|87.50%|[23]|
|:large_blue_circle:|com/starrocks/load/pipe/FilePipeSource.java|21|24|87.50%|[71, 150, 151]|
|:large_blue_circle:|com/starrocks/load/pipe/filelist/RepoCreator.java|39|44|88.64%|[44, 45, 80, 81, 82]|
|:large_blue_circle:|com/starrocks/load/pipe/Pipe.java|16|18|88.89%|[143, 388]|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|55|57|96.49%|[560, 561]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|27|28|96.43%|[2689]|
|:large_blue_circle:|com/starrocks/load/pipe/filelist/RepoExecutor.java|43|44|97.73%|[85]|
|:large_blue_circle:|com/starrocks/catalog/Database.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/PipeManager.java|19|19|100.00%|[]|
|:large_blue_circle:|com/starrocks/load/pipe/filelist/FileListRepo.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/system/information/InfoSchemaDb.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/pipe/AlterPipePauseResume.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/system/information/PipeFileSystemTable.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/PartitionInfo.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/statistic/StatisticsMetaManager.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/CatalogUtils.java|1|1|100.00%|[]|

mofeiatwork commented on fe/fe-core/src/main/java/com/starrocks/service/FrontendServiceImpl.java:
```diff
@@ -490,6 +503,87 @@ public TListMaterializedViewStatusResult listMaterializedViewStatus(TGetTablesPa
+        RepoAccessor repo = RepoAccessor.getInstance();
+        List<PipeFileRecord> files = repo.listAllFiles();
+        for (PipeFileRecord record : files) {
+            TListPipeFilesInfo file = new TListPipeFilesInfo();
```
good point. I will optimize it later.

mofeiatwork commented on fe/fe-core/src/main/java/com/starrocks/service/FrontendServiceImpl.java:
```diff
@@ -490,6 +503,87 @@ public TListMaterializedViewStatusResult listMaterializedViewStatus(TGetTablesPa
+        UserIdentity userIdentity = UserIdentity.fromThrift(params.getUser_ident());
+
+        PipeManager pm = GlobalStateMgr.getCurrentState().getPipeManager();
+        Map<PipeId, Pipe> pipes = pm.getPipesUnlock();
```
it's not. I will optimize it later.
